### PR TITLE
Add a comment to Packages.Data.Props

### PR DESCRIPTION
**Pull request by m-redding**
Adding a comment about PowerShell version conflicts in the Packages.Data.Props file. This contributes to https://github.com/Azure/azure-sdk-for-net/issues/40978

There isn't any specific guidance about this anywhere I could find that we could just link to. This information is from my own personal investigation after talking to a number of people, reading various GitHub issues, and testing things out.

### CODE REVIEW

**Comment by m-redding** [comment]
cc: @annelo-msft @christothes  @JoshLove-msft

File: eng/Packages.Data.props
Comment (christothes) [inline eng/Packages.Data.props:67]: Is there more detail we could add on how to check the dependencies?

File: eng/Packages.Data.props
Comment (JoshLove-msft) [inline eng/Packages.Data.props:64]: For Functions, this file can be referenced - https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json

Versions cannot be higher than those listed in the file.

Comment (m-redding) [reply eng/Packages.Data.props:64]: Added! Let me know if it looks good.

Comment (JoshLove-msft) [reply eng/Packages.Data.props:64]: Looks good!

**Review by JoshLove-msft — APPROVED** [review]

**Review by christothes — APPROVED** [review]
:shipit:

File: eng/Packages.Data.props
Comment (jsquire) [inline eng/Packages.Data.props:59]: Should we link to, or mention, the process for getting approved?

**Review by jsquire — APPROVED** [review]

File: eng/Packages.Data.props
Comment (m-redding) [reply eng/Packages.Data.props:59]: Would this work? https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-things-to-know I couldn't find anything better

File: eng/Packages.Data.props
Comment (JoshLove-msft) [inline eng/Packages.Data.props:66]: Does it have to be lower, or less than or equal to?

File: eng/Packages.Data.props
Comment (jsquire) [reply eng/Packages.Data.props:59]: I'd vote for just grabbing the email address and mentioning that if you need approval, that's who you talk to.   I'm afraid that it gets lost in all the words in the contributing guide.

File: eng/Packages.Data.props
Comment (m-redding) [reply eng/Packages.Data.props:66]: Ah yeah just less than or equal to, will fix

File: eng/Packages.Data.props
Comment (m-redding) [reply eng/Packages.Data.props:59]: updated

File: eng/Packages.Data.props
Comment (m-redding) [reply eng/Packages.Data.props:67]: Updated!

File: eng/Packages.Data.props
Comment (weshaggard) [inline eng/Packages.Data.props:60]: I don't think they should be contacting EngSys for these approvals. That should be handled by the language team or the architects depending on which reference we are talking about.

File: eng/Packages.Data.props
Comment (weshaggard) [inline eng/Packages.Data.props:71]: We have a set of deps files we use to validate our PS dependencies see https://github.com/Azure/azure-sdk-for-net/blob/main/eng/pipelines/aggregate-reports.yml#L84 as an example.

File: eng/Packages.Data.props
Comment (weshaggard) [reply eng/Packages.Data.props:60]: Looks like that was copied from the contributing which I also don't believe is correct.

File: eng/Packages.Data.props
Comment (jsquire) [reply eng/Packages.Data.props:71]: Those are outdated and do not reflect the current state of PowerShell.   Still trying to figure out who owns them and where they live because they need some love.

File: eng/Packages.Data.props
Comment (jsquire) [reply eng/Packages.Data.props:60]: Any idea why that's the address in the existing blurb in the contrib guide?  I agree that it doesn't seem right that it's EngSys.

File: eng/Packages.Data.props
Comment (m-redding) [reply eng/Packages.Data.props:71]: It looks like the file linked is using the dependencies from version 7.0.6 of PowerShell which was released in March 2021. Is there a reason to enforce compatibility with this version of PowerShell?
When I investigated PowerShell compatibility as part of https://github.com/Azure/azure-sdk-for-net/issues/33683 I talked to a couple people on the PowerShell team and looked at old issues (https://github.com/Azure/azure-sdk-for-net/issues/8739 / https://github.com/Azure/azure-sdk-for-net/issues/25824) in our repo and was told that we should validate compatibility with the latest version.

File: eng/Packages.Data.props
Comment (m-redding) [reply eng/Packages.Data.props:60]: I couldn't find any specific guidance for approval of dependencies in the docs/wiki/engineering hub, maybe the best point of contact is just our team? Maybe azsdknet@microsoft.com?

**Review by annelo-msft — APPROVED** [review]

File: eng/Packages.Data.props
Comment (jsquire) [reply eng/Packages.Data.props:60]: Let's drop the email entirely and just refer whomever is attempting to make the change to "seek architect approval."    We should put the Core team members as well as the architects as owners of the packages file in CODEOWNERS to make sure that things don't sneak in.

Comment (m-redding) [reply eng/Packages.Data.props:60]: That makes sense to me. I can make a separate PR to propose the update to CODEOWNERS

Comment (m-redding) [reply eng/Packages.Data.props]: Here is the PR : https://github.com/Azure/azure-sdk-for-net/pull/41039

Comment (weshaggard) [reply eng/Packages.Data.props]: >Any idea why that's the address in the existing blurb in the contrib guide?

Probably legacy and we should probably also update that.

**Pull request merged**
